### PR TITLE
Filter tests

### PR DESCRIPTION
**Pull request by coldestbeerintown**
Please read the documentation in the FilterIT class. Most of it is going to be deleted and it is there just for now, for you to read. The test class verifies the functionality of the filters as well as it confirms the failure of some of the filters.

### CODE REVIEW

File: src/main/java/com/telekom/m2m/cot/restsdk/users/UserApi.java [hunk 1]
```diff
@@ -29,6 +30,10 @@ public UserCollection getUsers(String tenant) {
         return new UserCollection(cloudOfThingsRestClient, tenant);
     }
 
+    public UserCollection getUsersWithFilters(FilterBuilder filter, String tenant) {
```
Comment (coldestbeerintown) [inline src/main/java/com/telekom/m2m/cot/restsdk/users/UserApi.java]: This method is added just for now, it is going to be deleted during merging. It is required to test filters with users.

File: src/main/java/com/telekom/m2m/cot/restsdk/util/Filter.java [hunk 2]
```diff
@@ -191,6 +191,31 @@ public FilterBuilder byUser(String user) {
         }
 
         /**
+         * Adds a build for an id.
```
Comment (coldestbeerintown) [inline src/main/java/com/telekom/m2m/cot/restsdk/util/Filter.java]: Below two methods are also going to be deleted, they are there just to check whether filters can be used with users or not.

**Comment by coldestbeerintown** [comment]
The parts that are going to be deleted before merging are as follows:
In Filter.java, the following methods were added only to test users:
-FilterBuilder byId(String id){...}
-FilterBuilder byUserName(String userName) {...}
They will be deleted.

In FilterIT.java:
public void testFiltersViaUsers() throws Exception {...}
will be deleted.

In UserApi.java:
public UserCollection getUsersWithFilters(FilterBuilder filter, String tenant) {...}
will be deleted.

Furthermore, the comments in the FilterIT will be trimmed.

File: src/test/java/com/telekom/m2m/cot/restsdk/util/FilterIT.java [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+		final Filter.FilterBuilder filterBuilder = Filter.build().byUserName("userNameForFilters");
+		UserCollection users = userApi.getUsersWithFilters(filterBuilder, tenant);
+
+		// Now, let's prove that this collection holds more than one user:
+		assertTrue(users.getUsers().length > 1);
+
+	}
+
+	@Test
+	public void testFiltersViaAlarms() throws Exception {
```
Comment (Andyck) [inline src/test/java/com/telekom/m2m/cot/restsdk/util/FilterIT.java]: imo this AlarmFilter IT are partially redundant and belongs into AlarmApiCollectionIT

File: src/test/java/com/telekom/m2m/cot/restsdk/util/FilterIT.java [hunk 4]
```diff
@@ -0,0 +1,313 @@
...
+		}
+
+		// The assertion verifies that in the array of alarms, some of them does not
+		// have the text we set.
+		assertFalse(allAlarmsWithSameText);
+
+	}
+
+	@Test
+	public void testFiltersViaEvents() throws Exception {
```
Comment (Andyck) [inline src/test/java/com/telekom/m2m/cot/restsdk/util/FilterIT.java]: This test belongs in EventApiCollectionIT I think

File: src/test/java/com/telekom/m2m/cot/restsdk/util/FilterIT.java (hunk 3 above)
Comment (coldestbeerintown) [reply src/test/java/com/telekom/m2m/cot/restsdk/util/FilterIT.java]: Yes these test can indeed be moved. The filters were used in different tests before. But they were never used just for the sake of testing specifically them. Therefore I thought such a class is necessary. But as long as the test does its job, it does not matter to me much where they reside.

File: src/test/java/com/telekom/m2m/cot/restsdk/util/FilterIT.java (hunk 4 above)
Comment (coldestbeerintown) [reply src/test/java/com/telekom/m2m/cot/restsdk/util/FilterIT.java]: This test is deleted as I saw that in fact all of the filters are used already in EventApiCollectionIT.

File: src/test/java/com/telekom/m2m/cot/restsdk/util/FilterIT.java (hunk 3 above)
Comment (coldestbeerintown) [reply src/test/java/com/telekom/m2m/cot/restsdk/util/FilterIT.java]: I now moved this test to AlarmApiCollectionIT.

File: src/main/java/com/telekom/m2m/cot/restsdk/users/UserApi.java [hunk 5]
```diff
@@ -5,6 +5,7 @@
 import com.google.gson.Gson;
 import com.telekom.m2m.cot.restsdk.CloudOfThingsRestClient;
 import com.telekom.m2m.cot.restsdk.util.ExtensibleObject;
+import com.telekom.m2m.cot.restsdk.util.Filter.FilterBuilder;
```
Comment (Andyck) [inline src/main/java/com/telekom/m2m/cot/restsdk/users/UserApi.java]: remove unused import

**Pull request merged**